src/ngraph/runtime/cpu/cpu_emitter.cpp

```diff
@@ -3952,6 +3953,45 @@ namespace ngraph
                 writer.block_end();
             }
 
+            template <>
+            void CPU_Emitter::EMITTER_DECL(ngraph::op::Dropout)
+            {
+                auto dropout = static_cast<const ngraph::op::Dropout*>(node);
+                if (out[0].get_element_type() != element::f64 &&
+                    out[0].get_element_type() != element::f32)
+                {
+                    throw ngraph_error("Unsupported element type");
+                }
+                size_t element_count = out[0].get_size();
+                size_t nthr = ngraph::runtime::cpu::executor::GetCPUExecutor().get_num_cores();
+                size_t chunk_size = (element_count + nthr - 1) / nthr;
+
+                writer.block_begin();
+                writer << "uint32_t seed = " << to_string(dropout->get_seed()) << ";\n";
+                writer << "double value = " << to_string(dropout->get_value()) << ";\n";
+
+                writer << "std::vector<std::minstd_rand> vmsr(" << to_string(nthr) << ");\n";
```

Comment:
this might lead to segfaults in codegen due to static variable initialization/destruction issues with MCJIT. Using states (similar to GenerateMask) and sticking this in there will solve the problem. For now, you could throw an unsupported exception here. Or change the kernel to make vmsr an optional input. If vmsr is provided, kernel uses it else it creates a new set of msrs internally and discards them.

Reply:
Ok, throwing an error for dropout codegen for now. Will work on support